### PR TITLE
Add class registry and graph model.

### PR DESCRIPTION
**Pull request by minasmart**

### CODE REVIEW

**Comment by minasmart** [comment]
@swalkinshaw would you mind taking a look at this?

File: src/graph-helpers/graph-model.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export default class GraphModel {
+  constructor(attrs) {
+    this.attrs = attrs;
+
+    Object.keys(this.attrs).filter(key => {
+      return !(key in this);
```
Comment (swalkinshaw) [inline src/graph-helpers/graph-model.js]: I'm never sure what the best solution is here but I don't know about silently just dropping these. Could easily lead to problems debugging. What about a message through `console.warn` (think that's non-standard)?

Comment (minasmart) [inline src/graph-helpers/graph-model.js]: The whole point is that you can override a getter with your own to fill in business logic. The point is for someone to be able to `class MyThing extends GraphModel`, and it builds out the proxy getters, and allows you to write your own special ones that rely on `attrs`

Comment (swalkinshaw) [inline src/graph-helpers/graph-model.js]: Okay that makes sense 👍

**Comment by swalkinshaw** [comment]
Looks good 🚀

**Pull request merged**